src/components/Toggle/components/Switch/Switch.js

```diff
@@ -57,16 +64,26 @@ const knobOnStyles = ({ theme, on }) =>
     );
   `;
 
-const SwitchKnob = styled('div')`
+const labelBaseStyles = () => css`
+  ${hideVisually()};
+`;
+
+const SwitchKnob = styled('span')`
   ${knobBaseStyles} ${knobOnStyles};
 `;
 
+// Important for accessibility
+const SwitchLabel = styled('span')`
+  ${labelBaseStyles};
+`;
+
 /**
  * A simple Switch component.
  */
 const Switch = ({ on, onToggle }) => (
   <SwitchTrack onClick={onToggle} on={on} role="switch" aria-checked={on}>
     <SwitchKnob {...{ on }} />
+    <SwitchLabel>{on ? 'on' : 'off'}</SwitchLabel>
```

Comment:
Ah! That with `hideVisually`. So someone on a screen reader would always get the current state of the switch? Would you mind making the strings props, so they can be passed in as translations?

Reply:
Exactly. Good point, I'll do that.